### DISCUSSION TITLE
Use multiple conditions inside AND operator in a nested relation

### DISCUSSION THREAD

**Discussion by ManishKarki1997** [prisma/prisma#8216] [Q&A] [answered]
Hi, its my first time using prisma and I'm stuck.
Tried asking in slack, but no luck.

Here's my stackoverflow question for this same problem
https://stackoverflow.com/questions/68326989/how-do-i-use-and-operator-with-multiple-query-parameters-in-nested-relation-wh

Here's the code that i'm trying to run

```ts
const products = await prisma.product.findMany({
    where: {
      AND: [
        {
          price: 549.99,
        },
        {
          filters: {
            some: {
              AND: [
                {
                  name: "RAM",
                  value: "32GB",
                },
                {
                   name: "Storage",
                   value: "1TB",
                 },

              ],
            },
          },
        },
      ],
    },
});

```

The idea is, products can be queried based on different parameters, like RAM, Storage, Graphics card etc. These params come dynamically from the frontend (using checkboxes and radio buttons etc). 
The above code works only if there is a single condition, e.g.
```ts
 some: {
     AND: [
          {
             name: "RAM",
             value: "32GB",
           },

      ],
  },
```

but, if i try to add another object, like in the topmost code (where I added storage condition), it always returns empty array, even though there are products matching these conditions.

My product schema looks like this (simplified)

```prisma
model Product {
  name                      String      
  filters                   Filter[]
}
```

My filter schema looks like this (simplified)

```prisma
model Fitler {
  name                        String      
  value                       String
  product                     Product?    @relation(fields: [productId], references:[id])
  productId                   Int?
}
```

Please let me know if you need any more details.
Thank you

**Comment by ryands17** [comment, marked as answer]
@ManishKarki1997 👋 

I have tested this with random data and this works:

```ts
await prisma.product.findMany({
      where: {
        AND: [
          { price: 21.99 },
          { filters: { some: { name: 'ram', value: '8GB' } } },
          { filters: { some: { name: 'storage', value: '256GB' } } },
        ],
      },
})
```

**Reply by ManishKarki1997** [reply]
@ryands17  Thank you very much, it works :) Quick question, are there any other platforms, like forums, to ask prisma related questions? I couldn't find any active ones. Is it just here? Thanks!

**Reply by ryands17** [reply]
You can ask here and on Slack. I guess the reason your question wasn't answered by anyone on Slack because it was a weekend 🙂 

Slack and GitHub discussions is super active if you check the activity there daily.

**Reply by inv-akhils** [reply]
Prisma will generate multiple `in clause` conditions, that is one for each **some** keyword. In the inner clause, it will join the Product and Filter table. Is there any Prisma option to get the result using a single `in clause`, like below

`> SELECT * from product where id in (SELECT productId from filter where (name = 'ram' and value = '8GB) or (name = 'storage' and value = '256GB') group by productId having count(productId) = 2)`

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
